google-cloud-bigtable/lib/google/cloud/bigtable.rb

```diff
@@ -69,18 +71,28 @@ module Bigtable
       def self.new \
           project_id: nil,
           credentials: nil,
+          emulator_host: nil,
           scope: nil,
           client_config: nil,
           timeout: nil
         project_id = (project_id || default_project_id).to_s
-
-        raise InvalidArgumentError, "project_id is required" unless project_id
+        raise ArgumentError, "project_id is required" if project_id.empty?
 
         scope ||= configure.scope
         timeout ||= configure.timeout
         client_config ||= configure.client_config
-        credentials ||= default_credentials(scope: scope)
+        emulator_host ||= configure.emulator_host
+        if emulator_host
+          return Bigtable::Project.new(
+            Bigtable::Service.new(
+              project_id, :this_channel_is_insecure,
+              host: emulator_host, timeout: timeout,
+              client_config: client_config
+            )
+          )
+        end
 
+        credentials ||= default_credentials(scope: scope)
```

Comment:
credentials are not required in emulator mode, so emulator users should not fail for empty credentials.

Reply:
Correct, if it is using the emulator the correct object will be returned above, on line 86.